contrib/babelfishpg_tsql/src/forjson.c

```diff
@@ -112,6 +118,15 @@ tsql_query_to_json_internal(const char *query, int mode, bool include_null_value
 		ereport(ERROR,
 				(errcode(ERRCODE_DATA_EXCEPTION),
 				 errmsg("invalid query")));
+	
+	if (SPI_processed==0)
+	{
+		SPI_finish();
+		return NULL;
+	}
+
+	// To check if query output table has columns with datatypes that are currently not supported in FOR JSON
+	tsql_unsupported_datatype_check();
```

Comment:
We should catch the exception and call `SPI_finish()`.

Reply:
Resolving as discussed.